### PR TITLE
Chapter 9

### PR DESCRIPTION
**Pull request by mirulzaei**

### CODE REVIEW

**Comment by mirulzaei** [comment]
ready

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -2,4 +2,5 @@ module ApplicationHelper
   def full_title page_title
     base_title = "Sample App"
     page_title.blank? ? base_title : "#{page_title} | #{base_title}"
+
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: you missing "end"

Comment (mirulzaei) [reply app/helpers/application_helper.rb]: i did put the end but it did not appear on the git

**Comment by mirulzaei** [comment]
ready

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -15,4 +15,4 @@
  */
  @import 'bootstrap-sprockets';
  @import 'bootstrap';
- @import 'color';
+ @import 'color';
```
Comment (chungpv-1008) [inline app/assets/stylesheets/application.scss]: add endline

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -12,6 +13,20 @@ def create
...
   def destroy
     log_out
+    log_out if logged_in?
     redirect_to root_url
   end
+
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by id: user_id
+    elsif (user_id = cookies.signed[:user_id])
```
Comment (chungpv-1008) [inline app/controllers/sessions_controller.rb]:
```suggestion
    if user_id = session[:user_id]
      @current_user ||= User.find_by id: user_id
    elsif user_id = cookies.signed[:user_id]
```

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -3,17 +3,40 @@ def log_in user
...
+    cookies.delete :remember_token
+  end
+
   def log_out
-    session.delete :user_id
+    forget(current_user)
+    session.delete(:user_id)
     @current_user = nil
-  end
+  end
```
Comment (chungpv-1008) [inline app/helpers/sessions_helper.rb:41]:
```suggestion
  end
```

File: app/views/sessions/new.html.erb [hunk 5]
```diff
@@ -7,8 +7,12 @@
...
+      <%= f.label :remember_me, class: "checkbox inline" do %>
+        <%= f.check_box :remember_me %>
+        <span>Remember me on this computer</span>
+      <% end %>
       <%= f.submit "Log in", class: "btn btn-primary" %>
     <% end %>
     <p><%= t "new_user?" %><%= link_to t("sign_up_now!"), signup_path %></p>
   </div>
-</div>
+</div>
```
Comment (chungpv-1008) [inline app/views/sessions/new.html.erb]: Add endline for this file

File: config/routes.rb [hunk 6]
```diff
@@ -8,4 +9,5 @@
   get "/login",to: "sessions#new"
   post "/login", to: "sessions#create"
   delete "/logout", to: "sessions#destroy"
+
```
Comment (chungpv-1008) [inline config/routes.rb]:
```suggestion
```

**Comment by mirulzaei** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**